### ISSUE TITLE
Inconsistent indentation allowed on chained if-else

### ISSUE THREAD

**Issue by whimatthew** [eslint/eslint#6956]
**What version of ESLint are you using?**
v3.3.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**
```js
{
  "env": {
    "node": true,
    "es6": true,
    "mocha": true
  },
  "rules": {
    "no-bitwise": 2,
    "curly": 0,
    "indent": [
      2,
      2,
      {
        "SwitchCase": 1
      }
    ],
    "no-caller": 2,
    "quotes": [
      2,
      "single"
    ],
    "no-undef": 2,
    "no-unused-vars": 2,
    "strict": 0,
    "max-len": [
      2,
      {
        "code": 100,
        "ignoreComments": true
      }
    ],
    "no-use-before-define": 0,
    "new-cap": 2,
    "no-sparse-arrays": 0,
    "no-trailing-spaces": 2
  },
  "parserOptions": {
    "sourceType": "module"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
```js
  if (type === '1') abacus.inc('rest.status.info');
    else if (type === '2') abacus.inc('rest.status.ok');
  else if (type === '4') abacus.inc('rest.status.clientError');
  else if (type === '5') abacus.inc('rest.status.serverError');
```
With the above if-else chain, I have the first else indented too far and we get the correct output: `error  Expected indentation of 2 space characters but found 4  indent`

But if the first else is correct, I find that all subsequent lines in the if-else chain don't generate errors when their indentation is off.

```js
  if (type === '1') abacus.inc('rest.status.info');
  else if (type === '2') abacus.inc('rest.status.ok');
               else if (type === '4') abacus.inc('rest.status.clientError');
      else if (type === '5') abacus.inc('rest.status.serverError');
```
This generates no output and linting passes.

**What did you expect to happen?**
I would expect the second input to generate indentation errors:
```sh
error  Expected indentation of 2 space characters but found 15  indent
error  Expected indentation of 2 space characters but found 6  indent
```

**What actually happened? Please include the actual, raw output from ESLint.**
Linting passes and generates no output.

**Comment by vitorbal** [comment]
Thanks for the report, @whimatthew! I can reproduce this. Seems like all else statements after the first one are not triggering an indentation error.

**Comment by not-an-aardvark** [comment]
Working on this.
